app/views/locations/_form.html.erb

```diff
@@ -1,60 +1,74 @@
-          <input id="geo_location" type="text" class="form-control"><br/>
-          <h4 class="font-size-small">Option 2 : By dragging the map</h4></br>
-          <h4 class="font-size-small">Option 3 : By giving co-ordinates</h4></br>
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+
+    <title>Leaflet.BlurredLocation</title>
+
+    <script src="https://maps.googleapis.com/maps/api/js?libraries=places&language=en&key=AIzaSyDWgc7p4WWFsO3y0MTe50vF4l4NUPcPuwE"></script>
+
+  </head>
+
+  <body>
+
+    <div class="container" >
+
+      <h3>Enter a location</h3>
+
+      <label>
+        <input id="browserLocation" type="checkbox" checked onchange='blurredLocation.geocodeWithBrowser(document.getElementById("browserLocation").checked);'>
+        Get my location
+      </label>
+      <br/>
+      <label>
+        <input id="obscureLocation" type="checkbox" checked onchange='blurredLocation.setBlurred(document.getElementById("obscureLocation").checked);'>
+        Blur my location
+      </label>
+
+      <hr />
+
+      <p><b>By place name</b></p>
+      <p><input id="placenameInput" type="text" class="form-control" /></p>
+
+      <p>
+        <b>By entering co-ordinates</b>
+        <div class="row">
           <div class="col-md-6">
-            <input id="lat" type="text" class="form-control" placeholder="Latitude"><br/>
+            <p><input id="lat" type="text" class="form-control" placeholder="Latitude"></p>
           </div>
           <div class="col-md-6">
-            <input id="lng" type="text" class="form-control" placeholder="Longitude"><br/>
+            <p><input id="lng" type="text" class="form-control" placeholder="Longitude"></p>
           </div>
-          <label class="h2 font-size-big"><input type="checkbox" onchange='getLocation(this);'>&nbsp;&nbsp;Obscure my exact location</label>
-        </div>
-        <div class="col-md-4">
-          <h3 class="font-size-small">On a map : </h3>
-          <%= render_map(17.4472, 78.3488, []) %>
-          <script>
-
-            addGrid(map);
-            geoLocateFromInput('geo_location');
-            geoLocateFromLatLng('lat','lng');
-
-          </script>
-          <h3 class="font-size-small">In Text:</h3><br/>
-          <input id="location" type="text" class="form-control"><br/>
         </div>
-      </div>
-</div>
-<script src="https://maps.googleapis.com/maps/api/js?libraries=places&language=en&key=AIzaSyDWgc7p4WWFsO3y0MTe50vF4l4NUPcPuwE"></script>
+      </p>
+
+      <p><b>By dragging the map</b></p>
+
+      <div id="map" class="leaflet-map" style="width: 100%; height: 400px;"></div>
+
+      <br />
+      <input editable="false" id="placenameDisplay" type="text" class="form-control" />
+
+    </div>
+
+    <script>
+
+      var options = {
+        InterfaceOptions: {
+          latId: 'lat',
+          lngId: 'lng'
+        }
+      }
+
+      var blurredLocation = new BlurredLocation(options);
+
+      blurredLocation.panMapToGeocodedLocation("placenameInput");
```

Comment:
Then here we'd want to create a button and click listener so you can save the location and it runs `addTag()` -- which I believe takes just the tagname as a parameter. But you can look at @shelbyfire's recent pull requests to confirm that usage.

Reply:
Er, @shelbyfirebaugh I meant!

Reply:
:+1: